Add tags field to Filestore Backups for TagsR2401

SamanthaMathews opened the pull request:
Add tags field to backup resource to allow setting tags on backup resources at creation time.
Part of b/364923942

release-note:none
```
filestore: added `tags` field to `filstore_backup` to allow setting tags for backups at creation time
```
```
The contents of this code are entirely owned by Google LLC in accordance with the agreement between Google LLC and the third party submitting this code into Google's open source repository
```

BBBmau commented:
could you fix the changelog so that it follows what's shown in the guidelines?

https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11614-old..auto-pr-11614) ( 4 files changed, 148 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11614-old..auto-pr-11614) ( 4 files changed, 148 insertions(+), 1 deletion(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-11614-old..auto-pr-11614) ( 1 file changed, 17 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_filestore_backup` (2 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_filestore_backup" "primary" {
  tags = # value needed
}


```

modular-magician commented:
#### Tests analytics
Total tests: 0
Passed tests: 0
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>filestore</li>

</ul>

</blockquote>
</details>
#### Non-exercised tests

Tests were added that are skipped in VCR:
- TestAccFilestoreBackup_tags
$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11614/artifacts/19750438-1472-4a4d-9bc6-b3b66a51b82d/build-log/replaying_test.log)

BBBmau commented on mmv1/third_party/terraform/services/filestore/resource_filestore_backup_test.go:
```diff
@@ -113,3 +142,87 @@ resource "google_filestore_backup" "backup" {
+          }
+	  tags = {`, context)
+
+	l := ""
```
There's a build error due to this line:
```hcl
google-beta/services/filestore/resource_filestore_backup_test.go:179:13: expected ')', found ','
```

SamanthaMathews replied:
Done

abd-goog commented on mmv1/third_party/terraform/services/filestore/resource_filestore_backup_test.go:
```diff
@@ -113,3 +139,41 @@ resource "google_filestore_backup" "backup" {
+          labels = {
+            "files":"label1",
+            "other-label": "label2"
+          }
```
Do not hard code tags in here, use the parameter provided. Example: https://screenshot.googleplex.com/BmcLcKPfbbjhbji

SamanthaMathews replied:
done

abd-goog commented on mmv1/third_party/terraform/services/filestore/resource_filestore_backup_test.go:
```diff
@@ -113,3 +114,72 @@ resource "google_filestore_backup" "backup" {
+	t.Parallel()
+
+        org := envvar.GetTestOrgFromEnv(t)
+	instName := fmt.Sprintf("tf-fs-inst-%d", acctest.RandInt(t))
```
nit: instanceName backupName instead of instName bkupName in all usages.

c2thorn commented:
Discussing with @SamanthaMathews offline about this PR and the go rewrite migration

NickElliot committed:
Convert from Ruby to Go

deletion_protection-backup-tags
commit terraform-filestore-backup-tags
commit terraform-filestore-backup-tags
commit terraform-filestore-backup-tags
commit terraform-filestore-backup-tags
commit terraform-filestore-backup-tag

NickElliot commented:
Converted, backed up at https://github.com/NickElliot/magic-modules/tree/terraform-filestore-backup-tags-11614-backup

SamanthaMathews committed:
Merge branch 'GoogleCloudPlatform:main' into terraform-filestore-backup-tags

SamanthaMathews committed:
Merge branch 'GoogleCloudPlatform:main' into terraform-filestore-backup-tags
